### PR TITLE
refine ContentType compare logic in Validation Build Rules

### PR DESCRIPTION
**Pull request by fkpwolf**
[AB#530134](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/530134)

when compare Error.File.ContentType & Rule.ContentType, if any type is null, then ignore content  type comparison.

For `uid-not-found` error, since the Learn Modules file got deleted, we have no idea of it's ContentType.

### CODE REVIEW

File: src/docfx/validation/CustomRuleProvider.cs [hunk 1]
```diff
@@ -122,9 +122,11 @@ private bool TryGetCustomRule(Error error, [MaybeNullWhen(false)] out CustomRule
...
                     var source = error.Source?.File;
                     var pageType = source != null ? _documentProvider.GetPageType(source) : null;
-                    if (rule.PropertyPath.Equals(error.PropertyPath, StringComparison.Ordinal) &&
-                        rule.ContentTypes != null &&
-                        rule.ContentTypes.Contains(pageType))
+                    var isPageTypeInScope =
+                        pageType == null ||
+                        rule.ContentTypes == null ||
+                        rule.ContentTypes.Contains(pageType);
+                    if (rule.PropertyPath.Equals(error.PropertyPath, StringComparison.Ordinal) && isPageTypeInScope)
```
Comment (yufeih) [inline src/docfx/validation/CustomRuleProvider.cs:129]: use `==` if it is ordinal string comparison.

```suggestion
                    if (rule.PropertyPath == error.PropertyPath && isPageTypeInScope)
```

File: src/docfx/validation/CustomRuleProvider.cs [hunk 2]
```diff
@@ -122,9 +122,11 @@ private bool TryGetCustomRule(Error error, [MaybeNullWhen(false)] out CustomRule
                     // compare with code + propertyPath + contentType
                     var source = error.Source?.File;
                     var pageType = source != null ? _documentProvider.GetPageType(source) : null;
-                    if (rule.PropertyPath.Equals(error.PropertyPath, StringComparison.Ordinal) &&
-                        rule.ContentTypes != null &&
-                        rule.ContentTypes.Contains(pageType))
+                    var isPageTypeInScope =
+                        pageType == null ||
+                        rule.ContentTypes == null ||
```
Comment (yufeih) [inline src/docfx/validation/CustomRuleProvider.cs:127]: Not need to check if `rule.ContentTypes == null` since it is marked as not nullable.

**Review by yufeih — APPROVED** [review]

File: src/docfx/validation/CustomRuleProvider.cs (hunk 1 above)
Comment (fkpwolf) [reply src/docfx/validation/CustomRuleProvider.cs]: ok

File: src/docfx/validation/CustomRuleProvider.cs (hunk 2 above)
Comment (fkpwolf) [reply src/docfx/validation/CustomRuleProvider.cs:127]: no, rule.ContentTypes is nullable

**Pull request merged**